packages/framework-provider-azure/src/library/events-store-adapter.ts

```diff
@@ -0,0 +1,140 @@
+    ...eventEnvelope,
+    createdAt: new Date().toISOString(),
+  }))
+}
+
+function toInputOperations(
+  eventEnvelopesChunkWithCreatedAt: Array<EventEnvelope>,
+  config: BoosterConfig
+): Array<CreateOperationInput> {
+  return eventEnvelopesChunkWithCreatedAt.map((eventEnvelopeWithCreatedAt) => {
+    const body: JSONObject = storableResource(config, eventEnvelopeWithCreatedAt)
+    return {
+      operationType: 'Create',
+      resourceBody: body,
+    }
+  })
+}
+
+async function batchEvents(
+  cosmosDb: CosmosClient,
+  config: BoosterConfig,
+  inputOperations: Array<CreateOperationInput>,
+  partitionKey: string
+): Promise<Response<OperationResponse[]>> {
+  const logger = getLogger(config, 'store-events-adapter#batchEvents')
+  try {
+    logger.debug('Storing EventEnvelopes with inputOperations:', inputOperations)
+    return await cosmosDb
+      .database(config.resourceNames.applicationStack)
+      .container(config.resourceNames.eventsStore)
+      .items.batch(inputOperations, partitionKey)
+  } catch (e) {
+    logger.error('Unexpected error storing events', e)
+    throw e
+  }
+}
+
+function groupByPartitionKey(
+  eventsPerPartitionKey: NonPersistedEventEnvelopePerPartitionKey,
+  event: NonPersistedEventEnvelope
+): NonPersistedEventEnvelopePerPartitionKey {
+  const partitionKey: string = partitionKeyForEvent(event.entityTypeName, event.entityID)
+  if (!eventsPerPartitionKey[partitionKey]) {
+    eventsPerPartitionKey[partitionKey] = []
+  }
+  eventsPerPartitionKey[partitionKey].push(event)
+  return eventsPerPartitionKey
+}
+
+/**
+ * Batch method expects a JSONObject. JSONObject valid types are: boolean | number | string | null | JSONArray | JSONObject
+ * This method tries to build a JSONObject from an EventEnvelope.
+ * It could fail if value contains a circular reference or a BigInt value is encountered.
+ * @param config
+ * @param eventEnvelope
+ */
+function storableResource(config: BoosterConfig, eventEnvelope: EventEnvelope): JSONObject {
+  const logger = getLogger(config, 'store-events-adapter#storableResource')
+  try {
+    const partitionKey = partitionKeyForEvent(eventEnvelope.entityTypeName, eventEnvelope.entityID)
+    const eventEnvelopeWithEntityID = {
+      ...eventEnvelope,
+      [eventsStoreAttributes.partitionKey]: partitionKey,
+      [eventsStoreAttributes.sortKey]: eventEnvelope.createdAt,
+    }
+    return JSON.parse(JSON.stringify(eventEnvelopeWithEntityID))
+  } catch (e) {
+    logger.error(`Could not parse eventEnvelope ${eventEnvelope} to JSONObject`, e)
+    throw e
+  }
+}
+
+/**
+ * Split events in chunks of size DEFAULT_CHUNK_SIZE
+ *
+ * @param arr
+ * @param size
+ */
+function chunkEvents(arr: Array<NonPersistedEventEnvelope>, size: number): Array<Array<NonPersistedEventEnvelope>> {
+  return Array.from({ length: Math.ceil(arr.length / size) }, (_: NonPersistedEventEnvelope, i: number) =>
+    arr.slice(i * size, i * size + size)
+  )
+}
```

Comment:
You can replace this function by using the lodash equivalent of `_.chunk(array, size)`

Reply:
We don't have lodash (and its type declarations) as a dependency. For just a single function that does exactly the same, I wouldn't bother adding another dependency, do you agree?

Reply:
Oh I though we had it. It's fine then